lib/membrane_h264_plugin/parser.ex

```diff
@@ -208,19 +192,91 @@ defmodule Membrane.H264.Parser do
     {[end_of_stream: :output], state}
   end
 
-  defp prepare_actions_for_aus(aus, pts, dts, state) do
-    Enum.reduce(aus, [], fn au, acc ->
-      sps_actions =
-        case Enum.find(au, &(&1.type == :sps)) do
-          nil ->
-            []
+  defp prepare_actions_for_aus(aus, state) do
+    prepare_actions_for_aus(aus, nil, nil, state)
+  end
 
-          sps_nalu ->
-            [stream_format: {:output, Format.from_sps(sps_nalu, framerate: state.framerate)}]
-        end
+  defp prepare_actions_for_aus(aus, buffer_pts, buffer_dts, state) do
+    {pts, dts} =
+      case state.mode do
+        :bytestream ->
+          if state.previous_timestamps == {nil, nil},
+            do: {-1, -1},
+            else: state.previous_timestamps
+
+        :nalu_aligned ->
+          state.previous_timestamps
+
+        :au_aligned ->
+          {buffer_pts, buffer_dts}
+      end
```

Comment:
I think we should put that clause inside of the `maybe_generate_timestamps`. I would also consider putting a "global" `au_counter` into the state which should simplify the logic. (more info in the general comment)